tests/resources/deeprec/movielens/ML-100K2Libffm.py

```diff
@@ -0,0 +1,209 @@
+import pandas as pd
+
+def create_dict(f_set):
+    f_dict = dict()
+    for i, j in enumerate(f_set):
+        f_dict[j] = i+1
+    return f_dict
+
+def create_userid_dict(filepath, field_num):
+    # feature_dict
+    f_set_list = []
+    for i in range(field_num):
+        f_set_list.append(set())
+    with open(filepath, "r", encoding="ISO-8859-1") as user:
+        for line in user:
+            line = line.strip().split("|")
+            for i in range(1, field_num + 1):
+                f_set_list[i-1].add(line[i])
+    f_dict_list = []
+    feature_num = 0
+    for i in range(field_num):
+        f_dict_list.append(create_dict(f_set_list[i]))
+        feature_num += len(f_dict_list[i])
+
+    user_feature_dict = dict()
+    with open(filepath, "r", encoding="ISO-8859-1") as user:
+        for line in user:
+            line = line.strip().split("|")
+            feature = ""
+            for i in range(1, field_num + 1):
+                feature = feature + " " + str(i) + ":" + str(f_dict_list[i-1][line[i]]) + ":1"
+            user_feature_dict[line[0]] = feature[1:]
+    return user_feature_dict, feature_num
+
+def create_itemid_dict(filepath, begin_fildnum):
+    # feature_dict
+    ym_set = set()
+    with open(filepath, "r", encoding="ISO-8859-1") as user:
+        for line in user:
+            line = line.strip().split("|")
+            release_date = line[2].split("-")
+            release_year_month = "-".join(release_date[1:])
+            ym_set.add(release_year_month)
+    ym_dict = create_dict(ym_set)
+    feature_num = len(ym_dict) + 19
+
+    item_feature_dict = dict()
+    with open(filepath, "r", encoding="ISO-8859-1") as user:
+        for line in user:
+            line = line.strip().split("|")
+            genre = line[-19:]
+            release_date = line[2].split("-")
+            release_year_month = "-".join(release_date[1:])
+            feature = str(begin_fildnum) + ":" + str(ym_dict[release_year_month]) + ":1"
+            for i in range(len(genre)):
+                if genre[i] == "1":
+                    feature = feature + " " + str(begin_fildnum+1) + ":" + str(i+1) + ":1"
+            item_feature_dict[line[0]] = feature
+    return item_feature_dict, feature_num
+
+
+
+
+
+def user_item_pairs(
```

Comment:
probably it would be nice to follow the same docstrings that we are using in the rest of the code. For example we are not using `item_df: object`. 

Also, a trick of the trade. There should be an intro before Args and Returns:
```
"""Get all pairs of users and items data.
    
Args:
        user_df (pd.DataFrame): User data containing unique user ids and maybe their features.
        item_df (pd.DataFrame): Item data containing unique item ids and maybe their features.
        user_col (str): User id column name.
        item_col (str): Item id column name.
        user_item_filter_df (pd.DataFrame): User-item pairs to be used as a filter.
        shuffle (bool): If True, shuffles the result.
    
Returns:
        pd.DataFrame: All pairs of user-item from user_df and item_df, excepting the pairs in user_item_filter_df
    """

```